CMakeLists.txt

```diff
@@ -234,19 +253,27 @@ endif ()
 
 function (zyan_add_test test)
     add_executable("Test${test}" "tests/${test}.cpp")
-
-    if (NOT MSVC)
-        target_compile_options("Test${test}" PRIVATE "-std=c++17")
-    endif ()
-
     target_link_libraries("Test${test}" "Zycore")
     target_link_libraries("Test${test}" "gtest")
-    set_target_properties("Test${test}" PROPERTIES FOLDER "Tests")
+    set_target_properties("Test${test}" PROPERTIES
+        LANGUAGE CXX
+        CXX_EXTENSIONS OFF
+        CXX_STANDARD_REQUIRED ON
+        FOLDER "Tests"
+    )
+    # CXX_STANDARD 17 is only available in CMake 3.8 and newer
+    if (CMAKE_VERSION VERSION_GREATER "3.7")
+        set_target_properties("Test${test}" PROPERTIES CXX_STANDARD 17)
+    elseif(NOT MSVC)
+        target_compile_options("Test${test}" PRIVATE "-std=c++17")
+    endif ()
```

Comment:
Feel free to just raise the minimum required version to `3.8`. Even the old-stable versions of both Ubuntu and Debian have newer packages these days, so I don't think we have to support even older stuff any longer. 

I'd also assume this to be broken on e.g. CMake `3.7.1`, because that would technically be greater than `3.7`? 

The documentation says:
> omitted components are treated as zero

Reply:
> Feel free to just raise the minimum required version to 3.8

Oh ok, I'll update the PR.

> I'd also assume this to be broken on e.g. CMake 3.7.1, because that would technically be greater than 3.7?

Yeah you're right, I missed that

Reply:
I think Gtest anyways started complaining about our old CMake version last time I configured/generated from scratch ^^

Reply:
> I think Gtest anyways started complaining about our old CMake version last time I configured/generated from scratch

That's because GTest 1.10 uses a deprecated minimum required version, see https://github.com/google/googletest/issues/3040